src/unix/signal.c

```diff
@@ -569,5 +571,6 @@ static void uv__signal_stop(uv_signal_t* handle) {
   uv__signal_unlock_and_unblock(&saved_sigmask);
 
   handle->signum = 0;
-  uv__handle_stop(handle);
+  if (handle->caught_signals == handle->dispatched_signals)
+    uv__handle_stop(handle);
```

Comment:
`uv__signal_loop_cleanup()` calls `uv__signal_stop()`. Can this change leave handles in limbo on shutdown?

Reply:
You're right as the signal pipe is closed there. Do you think adding a second argument `int do_handle_stop` to the `uv__signal_stop()` method to make sure `uv__handle_stop()` is called in that case would work?

Reply:
>  Can this change leave handles in limbo on shutdown?

I've been looking more into this and I *think* it should be fine as it is. Let me explain how I see it:
`uv__signal_loop_cleanup` is called on two places:
- On [`uv_loop_init()` failure](https://github.com/libuv/libuv/blob/v1.x/src/unix/loop.c#L104). This should not be a problem.
- On [`uv__loop_close()`](https://github.com/libuv/libuv/blob/v1.x/src/unix/loop.c#L147), which should be called from `uv_loop_close()`. But the function won't be called if the `uv_signal_t` handle is still in the `loop->handle_queue`. And the way that the handle can be removed from the queue is after `uv__make_close_pending()` has been called and that only happens when there's no pending signals in the signal pipe i.e.: when `handle->caught_signals == handle->dispatched_signals`.

I hope it makes some kind of sense. @bnoordhuis wdyt?

Reply:
I trust your analysis. :-)